scripts/SetupServiceFabric/SetupServiceFabric.sh

```diff
@@ -26,6 +32,105 @@ if [[ "xenial" != "$Distribution" && "bionic" != "$Distribution" ]]; then
     exit -1
 fi
 
+# Check ServiceFabricRuntimePath and ServiceFabricSdkPath have been provided. If yes, extract that
+ServiceFabricRuntimePath=""
+ServiceFabricSdkPath=""
+for i in "$@"
+do
+case $i in
+    -sfrt=*|--servicefabricruntime=*)
+    ServiceFabricRuntimePath="${i#*=}"
+    ;;
+    -sfs=*|--servicefabricsdk=*)
+    ServiceFabricSdkPath="${i#*=}"
+    ;;
+    *)
+    echo Error: Unknown option passed in: $i
+    exit 1
+    ;;
+esac
+done
+
+# Check if systemd is running as PID1, if not it should be enabled via systemd-genie
+pidone=$(ps --no-headers -o comm 1)
+if [[ "systemd" != "$pidone" ]]; then
+    # Set systemd-genie to run systemd as PID 1
+    echo "Setting up systemd-genie to run systemd as PID 1"
+    echo "Installing .NET SDK and runtime 5.0"
+    wget https://packages.microsoft.com/config/ubuntu/$(lsb_release -rs)/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+    dpkg -i packages-microsoft-prod.deb
+    rm packages-microsoft-prod.deb
+
+    echo "Installing the .NET SDK"
+    apt-get update; \
+    apt-get install -y apt-transport-https && \
+    apt-get update && \
+    apt-get install -y dotnet-sdk-5.0
+
+    echo "Installing the .NET runtime "
+    apt-get update; \
+    apt-get install -y apt-transport-https && \
+    apt-get update && \
+    apt-get install -y aspnetcore-runtime-5.0
+
+    echo "Adding the wsl-translinux repository"
+    apt install apt-transport-https
+
+    wget -O /etc/apt/trusted.gpg.d/wsl-transdebian.gpg https://arkane-systems.github.io/wsl-transdebian/apt/wsl-transdebian.gpg
+    chmod a+r /etc/apt/trusted.gpg.d/wsl-transdebian.gpg
+
+    file="/etc/apt/sources.list.d/wsl-transdebian.list"
+    echo "deb https://arkane-systems.github.io/wsl-transdebian/apt/ $(lsb_release -cs) main" >> $file
+    echo "deb-src https://arkane-systems.github.io/wsl-transdebian/apt/ $(lsb_release -cs) main" >> $file
+    cat $file
+
+    apt update
+    echo "Installing systemd-genie"
+    apt install -y systemd-genie
+
+    # Start genie
+    echo "Starting genie"
+    genie -i
+    echo "Genie has been started"
+fi
+
+#
+# If setup is being done inside WLS2 Distribution, then below helps in doing installation inside genie namespace if needed
+#
+genieCommand=''
+isGenieInstalled=$(apt list --installed | grep systemd-genie)
+
+if [[ ! -z "$isGenieInstalled" ]]; then
+    isGenieRunning=$(genie -r)
+    isOutsideGenie=$(genie -b)
+
+    if [[ "$isGenieRunning"=="runnning" && "$isOutsideGenie"=="outside" ]]; then
+        genieCommand="genie -c"
+    fi
+
+    # if genie is used for cluster management, current user should get permission to control service without sudo password.
+    # This enables cluster management from windows host via Powershell or LocalClusterManager.
```

Comment:
these things are required only if this is running inside genie context. Fix the checks above.

Reply:
if genie is running while setting up, then only update sudo privilege for the user.